x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -262,21 +245,20 @@ export class ExecuteReportTask implements ReportingTask {
     cancellationToken: CancellationToken,
     stream: Writable
   ): Promise<TaskRunResult> {
-    if (!this.taskExecutors) {
-      throw new Error(`Task run function factories have not been called yet!`);
-    }
+    const exportType = this.reporting.getExportTypesRegistry().getById(task.jobtype);
```

Comment:
It looks like previously I gave incorrect guidance about this, because the fields of the export type are kind of confusing.

```
// id; // used for exportTypeRegistry.get(); 
// jobType; // used in job params
// name: // used in user-facing content
```

Since the job type shouldn't be passed to `.getById`, this code should be:
```suggestion
    const exportType = this.exportTypesRegistry.get(({ jobType }) => jobType === task.jobtype);
```
I think this is the bug that is making reports not work currently.

Note: We should probably add a `getByJobType` method to the ExportTypeRegistry, since jobType is what comes in from task params: it seems more common to need to fetch an export type by `jobType` rather than `id`. (Doing this would simplify the `getJobContentEncoding` method as well - no more `taskExecutors` property will be needed)